Cargo.toml

```diff
@@ -1,14 +1,15 @@
 [workspace]
 members = [
-  "block-producer",
-  "node",
-  "proto",
-  "rpc",
-  "store",
-  "utils",
-  "test-macro",
-  "faucet"
+  "bin/node",
+  "crates/block-producer",
+  "crates/proto",
+  "crates/rpc",
+  "crates/faucet",
+  "crates/store",
+  "crates/utils",
+  "crates/test-macro",
 ]
```

Comment:
What is the motivation for creating this structure?

Also, if we go with it, shouldn't `faucet` go under `bin` too?

Reply:
It clearly separates the binary and it's components ( crates ) making the navigation of the repo clearer. This structured is inspired by RETH: https://github.com/paradigmxyz/reth

imo what I will do in this PR is to refactor the faucet to become a lib of the Node as with the other components ( block-producer, rpc, store ) and enable/disable the faucet using commang line arguments, EX: 

`cargo run start node --faucet`